tsp/tsp_client_response.py

```diff
@@ -47,3 +52,36 @@ def __init__(self, model, status, status_text):
 
         # The status message
         self.status_text = status_text
+
+        self.size = size
+    
+    def is_ok(self):
+        return self.status_code >= 200 and self.status_code < 400
+    
+    def __repr__(self):
+        return 'TspClientResponse(status_code={}, status_text={}, model={})'.format(self.status_code, self.status_text, self.model)
+
+class TspClientResponseEncoder(json.JSONEncoder):
+    
+    def default(self, obj):
+        if isinstance(obj, TspClientResponse):
+            # Convert TspClientResponse to a dictionary
+            if isinstance(obj.model, GenericResponse):
+                return {
+                    'model': GenericResponseEncoder().default(obj.model),
+                    RESPONSE_STATUS_KEY : obj.status,
+                    STATUS_MESSAGE_KEY: obj.status_text
+                }
+            elif isinstance(obj.model, OutputDescriptorSet):
+                return {
+                    'model': [OutputDescriptorEncoder().default(output_descriptor) for output_descriptor in obj.model.descriptors],
+                    RESPONSE_STATUS_KEY: obj.status,
+                    STATUS_MESSAGE_KEY: obj.status_text
+                }
+            else:
+                return {
+                    'model': obj.model,
```

Comment:
Some modules don't have a encoder implemented yet. For example, when running `--list-experiments` and do a json.dumps on it then I get an error:

```
TypeError: Object of type ExperimentSet is not JSON serializable
```

The same for `list-traces`:

```
TypeError: Object of type TraceSet is not JSON serializable
```

Reply:
I added those encoders and added them to the TspClientResponse.